src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -786,6 +893,11 @@ private uint GetParameterValEscape(ParameterSymbol parameter)
         {
             if (UseUpdatedEscapeRules)
             {
+                if (parameter.RefKind == RefKind.Out && parameter.EffectiveScope != DeclarationScope.Unscoped)
```

Comment:
Specification says

> A `out` parameter for a `ref struct` will have *safe-to-escape* of *return only*.

i.e. we need to check if parameter's type is ref struct here. Also, it seems like this STE is used on `out` parameters regardless of presence of `scoped` or `[UnscopedRef]` modifiers. So it seems like the check of `parameter.EffectiveScope` here can be deleted.